FuelLabs/fuel-indexer
# bugfix: `stop-previous` flag to remove wasm binary by `index_id`

Closes #786 

Note that the changes to the plugin side were already merged to master, and then reverted. The changes to the plugin are on this branch, but the diff doesn't show because of the previous merge. 

## Changelog
• Adds conditional logic based on query param to `remove_indexer`.
• Adds `remove_previous_indexer` queries function.
• Adds metrics for new query function.  

## Testing 
1. Build the new plugin binary `cargo build -p forc-index -p fuel-indexer --release --features fuel-core-lib`.
2. Refresh the DB and rebuild test binaries. 
3. Start local fuel-node.
4. Start the indexer.
5. Deploy the block explorer `./target/release/forc-index deploy --path examples/block-explorer/explorer-indexer `
6. Run the SQL query `select * from index_asset_registry_manifest;` 
   - Make a mental note of the single wasm binary in the DB. 👀 
7. Make some small change in the `lib.rs` of the block explorer.
8. Deploy the block explore with the new flag `./target/release/forc-index deploy --stop-previous --path examples/block-explorer/explorer-indexer`
 9. Run the same SQL query `select * from index_asset_registry_manifest;`
 
You should see that there is still only one wasm binary with the same index_id of 1. The old one was removed.
